avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailProjectTemplate.kt

```diff
@@ -0,0 +1,185 @@
+        configureAvailStdLib(config, lib)
+      }
+
+      updateBackingProjectFile(projectFilePath, config)
+    }
+  }
+
+  private fun AvailProjectV1.updateBackingProjectFile(
+    projectFilePath: String,
+    config: Config
+  ) {
+    if (projectFilePath.isNotEmpty()) {
+      // Update the backing project file.
+      val writer = JSONWriter.newPrettyPrinterWriter()
+      writeTo(writer)
+      File(projectFilePath).writeText(writer.contents())
+      config.environmentSettings.add(this, projectFilePath)
+    }
+  }
+
+  private fun AvailProjectV1.configureAvailStdLib(
+    config: Config,
+    lib: File
+  ) {
+    val libName = config.libraryName ?: AVAIL_STDLIB_ROOT_NAME
+    val libConfigDir = AvailEnvironment.projectRootConfigPath(
+      config.fileName, libName, config.projectLocation
+    )
+    optionallyInitializeConfigDirectory(libConfigDir)
+    val jar = AvailArtifactJar(lib.toURI())
+
+    val rootManifest = jar.manifest.roots[AVAIL_STDLIB_ROOT_NAME]
+    var sg =
+      importStyles(config, jar)
+    var tg =
+      importTemplates(config, jar)
+    val extensions = mutableListOf<String>()
+    var description = ""
+    rootManifest?.let {
+      sg = it.styles
+      tg = it.templates
+      extensions.addAll(it.availModuleExtensions)
+      description = it.description
+    }
+    val stdLib = AvailProjectRoot(
+      libConfigDir,
+      config.projectLocation,
+      libName,
+      AvailLibraries(
+        "org/availlang/${lib.name}",
+        Scheme.JAR,
+        AVAIL_STDLIB_ROOT_NAME
+      ),
+      LocalSettings(
+        AvailEnvironment.projectRootConfigPath(
+          config.fileName,
+          libName,
+          config.projectLocation
+        )
+      ),
+      sg,
+      tg,
+      extensions
+    )
+    stdLib.description = description
+    stdLib.saveLocalSettingsToDisk()
+    stdLib.saveTemplatesToDisk()
+    stdLib.saveStylesToDisk()
+    addRoot(stdLib)
+  }
+
+  private fun importTemplates(
+    config: Config,
+    jar: AvailArtifactJar
+  ) = if (config.importTemplates) {
+    jar.manifest.templatesFor(AVAIL_STDLIB_ROOT_NAME)
+      ?: TemplateGroup()
+  } else {
+    TemplateGroup()
+  }
+
+  private fun importStyles(
+    config: Config,
+    jar: AvailArtifactJar
+  ) = if (config.importStyles) {
+    jar.manifest.stylesFor(AVAIL_STDLIB_ROOT_NAME)
+      ?: StylingGroup()
+  } else {
+    StylingGroup()
+  }
+
+  private fun availProjectRoot(
+    rootConfigDir: String,
+    config: Config,
+    rootName: String,
+    rootLocationDir: String
+  ) = AvailProjectRoot(
+    rootConfigDir,
+    config.projectLocation,
+    rootName,
+    ProjectHome(
+      rootLocationDir,
+      Scheme.FILE,
+      config.projectLocation,
+      null
+    ),
+    LocalSettings(rootConfigDir),
+    StylingGroup(),
+    TemplateGroup()
+  )
+
+  private fun rootsLocationDir(rootsDir: String, rootName: String) = if (rootsDir.isNotEmpty()) {
+    "$rootsDir/$rootName"
+  } else {
+    rootName
+  }
```

Comment:
I chose your shorter style for the last one since it had fewer code in the branches. https://github.com/AvailLang/Avail/pull/310/commits/856dba8ff86587e9045a08d0ce08bb7928811895